test/functional/ex_cmds/wviminfo_spec.lua

```diff
@@ -0,0 +1,62 @@
+local helpers, lfs = require('test.functional.helpers'), require('lfs')
+local clear, execute, eq, neq, spawn, nvim_prog, set_session, wait =
+  helpers.clear, helpers.execute, helpers.eq, helpers.neq, helpers.spawn,
+  helpers.nvim_prog, helpers.set_session, helpers.wait
+
+-- Lua does not have a sleep function so we use the system command.  If the
+-- command does not support sub second precision we use math.floor() to get
+-- full seconds.
+local sleep = function(millisec)
+  local sec = millisec / 1000
+  local round = math.floor(sec)
+  if round == 0 then round = 1 end
+  os.execute('sleep '..sec..' || sleep '..round)
+end
+
+describe(':wviminfo', function()
+  local file = 'foo'
+  before_each(function()
+    clear()
+    os.remove(file)
+  end)
+
+  it('creates a file', function()
+    -- TODO
+    -- Set up the nvim session to be able to write viminfo files.  Is it
+    -- possible to do this outside of the it() call?
```

Comment:
Since clear() sets up a new session, doing it outside of the it() call wouldn't work. That's why I can `set_session` a hack--ideally, session.lua should support multiple instances. Currently it does not (it's a minor refactoring to do so, just haven't got round to it).